console/Script.cs

```diff
@@ -1,90 +1,114 @@
+                var tables = db.FindTablesRegEx(DataTablesPattern);
+                foreach (var t in tables.Where(t => !db.DataTables.Contains(t))) {
+                    db.DataTables.Add(t);
+                }
+            }
 
-			Log(TraceLevel.Info, Environment.NewLine + "Snapshot successfully created at " + db.Dir);
-			var routinesWithWarnings = db.Routines.Select(r => new {
-				Routine = r,
-				Warnings = r.Warnings().ToList()
-			}).Where(r => r.Warnings.Any()).ToList();
-			if (routinesWithWarnings.Any()) {
-				Log(TraceLevel.Info, "With the following warnings:");
-				foreach (
-					var warning in
-						routinesWithWarnings.SelectMany(
-							r =>
-								r.Warnings.Select(
-									w => string.Format("- {0} [{1}].[{2}]: {3}", r.Routine.RoutineType, r.Routine.Owner, r.Routine.Name, w)))) {
-					Log(TraceLevel.Warning, warning);
-				}
-			}
-			return 0;
-		}
+            db.ScriptToDir(TableHint, Log);
 
-		private static void HandleDataTables(Database db, string tableNames) {
-			foreach (var value in tableNames.Split(',')) {
-				var schema = "dbo";
-				var name = value;
-				if (value.Contains(".")) {
-					schema = value.Split('.')[0];
-					name = value.Split('.')[1];
-				}
-				var t = db.FindTable(name, schema);
-				if (t == null) {
-					Console.WriteLine(
-						"warning: could not find data table {0}.{1}",
-						schema, name);
-				}
-				if (db.DataTables.Contains(t)) continue;
-				db.DataTables.Add(t);
-			}
-		}
-	}
+            Log(TraceLevel.Info, Environment.NewLine + "Snapshot successfully created at " + db.Dir);
+            var routinesWithWarnings = db.Routines.Select(r => new {
+                Routine = r,
+                Warnings = r.Warnings().ToList()
+            }).Where(r => r.Warnings.Any()).ToList();
+            if (routinesWithWarnings.Any()) {
+                Log(TraceLevel.Info, "With the following warnings:");
+                foreach (
+                    var warning in
+                        routinesWithWarnings.SelectMany(
+                            r =>
+                                r.Warnings.Select(
+                                    w =>
+                                        string.Format("- {0} [{1}].[{2}]: {3}", r.Routine.RoutineType, r.Routine.Owner,
+                                            r.Routine.Name, w)))) {
+                    Log(TraceLevel.Warning, warning);
+                }
+            }
+            return 0;
+        }
+
+        private List<string> HandleFilteredTypes() {
+            var filteredTypes = FilterTypes == null ? new List<string>() : FilterTypes.Split(',').ToList();
+
+            foreach (var filterType in filteredTypes) {
+                if (!Database.Dirs.Contains(filterType)) {
+                    Log(TraceLevel.Warning, string.Format("{0} is not a valid type.", filterType));
+                    Log(TraceLevel.Warning, string.Format("Valid types: {0}", Database.ValidTypes));
```

Comment:
This log is inside the foreach loop, seems kinda strange to keep outputting the same thing.  Can it be moved outside the loop?